yargs.js

```diff
@@ -345,19 +344,30 @@ function Yargs (processArgs, cwd, parentRequire) {
   self.pkgConf = function (key, path) {
     var conf = null
 
-    var obj = readPkgUp.sync({
-      cwd: path || requireMainFilename(parentRequire || require)
-    })
+    var obj = pkgUp(path)
 
     // If an object exists in the key, add it to options.configObjects
-    if (obj.pkg && obj.pkg[key] && typeof obj.pkg[key] === 'object') {
-      conf = obj.pkg[key]
+    if (obj[key] && typeof obj[key] === 'object') {
+      conf = obj[key]
       options.configObjects = (options.configObjects || []).concat(conf)
     }
 
     return self
   }
 
+  var pkg = null
+  function pkgUp (path) {
+    if (pkg && !path) return pkg
```

Comment:
There's a slight bug with this logic. If `pkgUp` is called first with a `path` (as in `.pkgConf('kewl', process.cwd())`) and then subsequently without a path (as in `.version()`), the parsed pkg object returned will be from the call with the path, which is most likely the wrong package.json file.

To fix, we should probably cache a map of path -> pkg object and return the cached object that matches the given path (normalized to `'__no_path__'` or something similar).